R/check.R

```diff
@@ -413,30 +413,25 @@ checkPlots <- function(plots) {
 checkMapping <- function(mapping) {
   checkList(mapping)
 
-  # stop if mapping object has less than 2 elements
-  if (length(mapping) > 0) stopifnot(length(mapping) > 1)
-  else return(NULL)
-
-  # check if list elements have the same size. If not, fill difference with NA.
-  listMaxLength <- max(sapply(mapping, length))
-  mapping <- lapply(lapply(mapping, unlist), "length<-", listMaxLength)
-
-  mappingdf <- as.data.frame(mapping, stringsAsFactors = FALSE)
-
-  # NAs are accepted, but not if all values for a model are NA
-  stopifnot(vapply(mappingdf, is.character, logical(1)))
-
-  # check if any given model has at least one feature aligned with another model
-  for (i in seq_along(mappingdf)) {
-    tempModel   <- mappingdf[!is.na(mappingdf[[i]]),]
-    if (nrow(tempModel) > 1) {
-      featAligned <- any(apply(!sapply(tempModel, is.na), 1, sum) > 1)
-    } else {
-      featAligned <- any(sum(!sapply(tempModel, is.na)) > 1)
-    }
-
-    if (!is.na(featAligned) && !featAligned) {
-      stop(sprintf("Model \"%s\" does not present any feature mapping to another model.", colnames(mappingdf)[[i]]))
+  for (i in seq_along(mapping)) {
+    # stop if mapping object contains anything else besides data frames
+    stopifnot("mapping object must be a list of at least one data frame" = inherits(mapping[[i]], "data.frame"))
```

Comment:
This new feature to customize the error messages was added in [R 4.0.0](https://cran.r-project.org/doc/manuals/r-release/NEWS.html):

> `stopifnot()` now allows customizing error messages via argument names, thanks
> to a patch proposal by Neal Fultz in PR#17688.

To support previous versions of R, please change these 3 instances of `stopifnot()` to instead to use the pattern:

```R
if (!) {
  stop()
}
```

eg

```R
if (!inherits(mapping[[i]], "data.frame")) {
  stop("mapping object must be a list of at least one data frame")
}
```

Reply:
Updated accordingly.